sentry-android-replay/src/main/java/io/sentry/android/replay/screenshot/CanvasStrategy.kt

```diff
@@ -0,0 +1,890 @@
+    // TODO implement
+  }
+
+  override fun drawText(text: CharArray, index: Int, count: Int, x: Float, y: Float, paint: Paint) {
+    paint.getTextBounds(text, index, count, tmpRect)
+    drawMaskedText(paint, x, y)
+  }
+
+  override fun drawText(text: String, x: Float, y: Float, paint: Paint) {
+    paint.getTextBounds(text, 0, text.length, tmpRect)
+    drawMaskedText(paint, x, y)
+  }
+
+  override fun drawText(text: String, start: Int, end: Int, x: Float, y: Float, paint: Paint) {
+    paint.getTextBounds(text, start, end, tmpRect)
+    drawMaskedText(paint, x, y)
+  }
+
+  override fun drawText(
+    text: CharSequence,
+    start: Int,
+    end: Int,
+    x: Float,
+    y: Float,
+    paint: Paint,
+  ) {
+    paint.getTextBounds(text, 0, text.length, tmpRect)
+    drawMaskedText(paint, x, y)
+  }
+
+  override fun drawTextOnPath(
+    text: CharArray,
+    index: Int,
+    count: Int,
+    path: Path,
+    hOffset: Float,
+    vOffset: Float,
+    paint: Paint,
+  ) {
+    // TODO implement
+  }
+
+  override fun drawTextOnPath(
+    text: String,
+    path: Path,
+    hOffset: Float,
+    vOffset: Float,
+    paint: Paint,
+  ) {
+    // TODO implement
+  }
+
+  override fun drawTextRun(
+    text: CharArray,
+    index: Int,
+    count: Int,
+    contextIndex: Int,
+    contextCount: Int,
+    x: Float,
+    y: Float,
+    isRtl: Boolean,
+    paint: Paint,
+  ) {
+    paint.getTextBounds(text, 0, index + count, tmpRect)
+    drawMaskedText(paint, x, y)
+  }
+
+  override fun drawTextRun(
+    text: CharSequence,
+    start: Int,
+    end: Int,
+    contextStart: Int,
+    contextEnd: Int,
+    x: Float,
+    y: Float,
+    isRtl: Boolean,
+    paint: Paint,
+  ) {
+    paint.getTextBounds(text, start, end, tmpRect)
+    drawMaskedText(paint, x, y)
+  }
+
+  override fun drawTextRun(
+    text: MeasuredText,
+    start: Int,
+    end: Int,
+    contextStart: Int,
+    contextEnd: Int,
+    x: Float,
+    y: Float,
+    isRtl: Boolean,
+    paint: Paint,
+  ) {
+    text.getBounds(start, end, tmpRect)
+    drawMaskedText(paint, x, y)
+  }
+
+  private fun sampleBitmapColor(bitmap: Bitmap, paint: Paint?, region: Rect?): Int {
+    if (bitmap.isRecycled) {
+      return Color.BLACK
+    }
+
+    val cache = bitmapColorCache[bitmap]
+    if (cache != null && cache.first == bitmap.generationId) {
+      return cache.second
+    } else {
+      singlePixelCanvas.drawBitmap(bitmap.asShared(), region, singlePixelBitmapBounds, paint)
+      val color = singlePixelBitmap.getPixel(0, 0)
+      bitmapColorCache[bitmap] = Pair(bitmap.generationId, color)
+      return color
+    }
+  }
+
+  private fun drawMaskedText(paint: Paint, x: Float, y: Float) {
+    textPaint.colorFilter = paint.colorFilter
+    val color = paint.color
+    textPaint.color = Color.argb(100, Color.red(color), Color.green(color), Color.blue(color))
+    tmpRect.offset(x.roundToInt(), y.roundToInt())
+    drawRect(tmpRect, solidPaint)
```

Comment:
oh btw, do you wanna use `drawRoundRect` so it aligns with the PixelCopy strategy? Same for images here